@here/harp-mapview/lib/FrustumIntersection.ts

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OrientedBox3 } from "@here/harp-geometry";
+import { MathUtils, Projection, ProjectionType, TileKey, TilingScheme } from "@here/harp-geoutils";
+import { assert } from "chai";
+import * as THREE from "three";
+import { CalculationStatus, ElevationRangeSource } from "./ElevationRangeSource";
+import { MapTileCuller } from "./MapTileCuller";
+import { MapViewUtils, TileOffsetUtils } from "./Utils";
+
+/**
+ * Represents a unique TileKey and the area it takes up on screen.
+ *
+ * Note, in certain tiling projections, it is possible to have an offset, which represents a tile
+ * which has fully wrapped around, hence this defaults to 0 to simplify usage for projections which
+ * don't require it.
+ */
+export class TileKeyEntry {
+    constructor(public tileKey: TileKey, public area: number, public offset: number = 0) {}
+}
+
+function getGeoBox(tilingScheme: TilingScheme, childTileKey: TileKey, offset: number) {
+    const geoBox = tilingScheme.getGeoBox(childTileKey);
+    const longitudeOffset = 360.0 * offset;
+    geoBox.northEast.longitude += longitudeOffset;
+    geoBox.southWest.longitude += longitudeOffset;
+    return geoBox;
+}
+
+namespace FrustumIntersection {
+    export interface Result {
+        /**
+         * Tiles intersected by the frustum.
+         */
+        readonly tileKeyEntries: Map<number, TileKeyEntry>;
```

Comment:
It isn't apparent what the key is, I guess morton code?

Reply:
I'll add a comment.